templates/demo/web3js-react-dapp-demo/src/AccountDetail.tsx

```diff
@@ -0,0 +1,283 @@
+          );
+
+          return next;
+        });
+      })
+      .on("receipt", (data) => {
+        setTransactions((prev: Map<string, string>) => {
+          const next = new Map<string, string>(prev);
+          next.set(
+            transactionId,
+            `${amount} wei sent to ${to} [Hash: ${data.transactionHash} Block #: ${data.blockNumber}]`,
+          );
+
+          return next;
+        });
+      })
+      .on("confirmation", (data) => {
+        const numConfirmations: bigint = data.confirmations;
+        const receipt = data.receipt;
+        setTransactions((prev: Map<string, string>) => {
+          const next = new Map<string, string>(prev);
+          next.set(
+            transactionId,
+            `${amount} wei sent to ${to} [Hash: ${receipt.transactionHash} Block #: ${receipt.blockNumber} Confirmations: ${numConfirmations}]`,
+          );
+
+          return next;
+        });
+
+        if (numConfirmations > 5) {
+          transferEvent.removeAllListeners();
+          transferEvents.current.delete(transactionId);
+        }
+      })
+      .on("error", (data) => {
+        setTransactions((prev: Map<string, string>) => {
+          const next = new Map<string, string>(prev);
+          next.set(
+            transactionId,
+            `Error sending ${amount} wei to ${to}: ${data}`,
+          );
+
+          return next;
+        });
+
+        transferEvent.removeAllListeners();
+        transferEvents.current.delete(transactionId);
+      });
+
+    transferEvents.current.set(transactionId, transferEvent);
+
+    transferEvent.catch((error: any) => {
+      transferEvent.removeAllListeners();
+      transferEvents.current.delete(transactionId);
+      if (error.code === 100) {
+        // user rejected transaction
+        setTransactions((prev: Map<string, string>) => {
+          const next: Map<string, string> = new Map(prev);
+          next.delete(transactionId);
+          return next;
+        });
+      } else {
+        setTransactions((prev: Map<string, string>) => {
+          const next = new Map<string, string>(prev);
+          next.set(
+            transactionId,
+            `Error sending ${amount} wei to ${to}: ${error}`,
+          );
+
+          return next;
+        });
+      }
+    });
+  }
+
+  return (
+    <>
+      <div>{address}</div>
+      <div>Balance in wei: {`${balance}`}</div>
+      <form onSubmit={transfer}>
+        <label>
+          Transfer to:{" "}
+          <input
+            value={transferTo}
+            onChange={transferFormChange}
+            name="to"
+            type="text"
+          />
+        </label>
+
+        <span style={{ paddingRight: "5px" }}></span>
+
+        <label>
+          Transfer amount in wei:{" "}
+          <input
+            value={!isNaN(transferAmount) ? transferAmount : ""}
+            onChange={transferFormChange}
+            name="amount"
+            type="number"
```

Comment:
Nitpick:  I feel like most ux's use ether as a measurement instead of wei as its easier to deal with instead of counting all 18 of the 0's